org.jacoco.core.test/src/org/jacoco/core/internal/analysis/MethodAnalyzerTest.java

```diff
@@ -43,7 +47,30 @@
 	@Before
 	public void setup() {
 		nextProbeId = 0;
-		method = new MethodNode();
+		method = new MethodNode(InstrSupport.ASM_API_VERSION) {
```

Comment:
@Godin Have you considered using ASM's `CheckMethodAdapter` instead? When you emit the method node to a instance of this adapter exactly this false API usage should be detected.